protobuf/v1/replica.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "common.proto";
+
+package mayastor.v1;
+
+service ReplicaRpc {
+  // Replica related methods.
+  //
+  // Replica allocates space from storage pool.
+  rpc CreateReplica (CreateReplicaRequest) returns (Replica) {}
+  rpc DestroyReplica (DestroyReplicaRequest) returns (google.protobuf.Empty) {}
+  rpc ListReplicas (ListReplicasOptions) returns (ListReplicasResponse) {}
+  rpc ShareReplica (ShareReplicaRequest) returns (ShareReplicaResponse) {}
+}
+
+message CreateReplicaRequest {
+  string name = 1;  // name of the replica
+  string uuid = 2;  // uuid of the replica
+  string pool = 3;  // name of the pool
+  PoolType pooltype = 4; // type of the pool
+  uint64 size = 5;  // size of the replica in bytes
+  bool thin = 6;    // thin provisioning
+  ShareProtocol share = 7;  // protocol to expose the replica over
+}
+
+// Destroy replica arguments.
+message DestroyReplicaRequest {
+  string uuid = 1;  // uuid of the replica
+}
+
+// Replica properties
+message Replica {
+  string name = 1;  // name of the replica
+  string uuid = 2;  // uuid of the replica
+  string pool = 3;  // name of the pool
+  PoolType pooltype = 4; // type of the pool
```

Comment:
Its a property of the pool. But having this here, helps to easily identify replicas that are on a specific type of pool. Otherwise, we will have to first fetch the replicas, then find the pooltype of each pools on which the replica is present by having a `ListPool` call .

Reply:
Removed pooltype from replica